File: kotlin/kotlin-spring-boot/src/main/kotlin/com/meridal/examples/kotlinspringboot/service/RecordingService.kt
package com.meridal.examples.kotlinspringboot.service

import com.meridal.examples.kotlinspringboot.model.Recording
import com.meridal.examples.kotlinspringboot.repository.RecordingRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service

@Service
class RecordingService @Autowired constructor(private val repository: RecordingRepository) {

    fun getRecording(id: Long) = repository.findByIdOrNull(id)

    fun getAllRecordings() = repository.findAll()

    fun saveRecording(recording: Recording) = repository.save(recording)

    fun deleteRecording(id: Long): Recording? {
        val recording = getRecording(id)
        if (recording != null) {
            repository.deleteById(id)
        }
        return recording
    }

}

File: scala/scala-spring-boot/src/main/scala/com/meridal/examples/scalaspringboot/service/RecordingService.scala
package com.meridal.examples.scalaspringboot.service

import org.springframework.beans.factory.annotation.Autowired
import com.meridal.examples.scalaspringboot.repository.RecordingRepository
import org.springframework.stereotype.Service
import com.meridal.examples.scalaspringboot.model.Recording

import scala.collection.mutable.ListBuffer
import scala.collection.JavaConverters._


@Service
class RecordingService(@Autowired val repository: RecordingRepository) {
  
  def getRecording(id: Long): Recording = repository.findById(id).orElse(null)

  def getAllRecordings(): List[Recording] = ListBuffer.from(repository.findAll().asScala).result()

  def saveRecording(recording: Recording) = repository.save(recording)

  def deleteRecording(id: Long): Recording = {
    val recording = getRecording(id)
    if (recording != null) {
      repository.deleteById(id)
    }
    return recording
  }
}
